src/Components/sharedComponents/RenderComponent.tsx

```diff
@@ -0,0 +1,18 @@
+import { ReactNode } from 'react';
+
+// Create a usable component to conditionally render react components. This
+// is especially useful when there are multiple components that need to be
+// rendered conditionally and improves the readability of the code.
+export const RenderComponent = ({
```

Comment:
woah i like this

Reply:
Yeah I was on the fence about this because it would make this PR huge

Reply:
I took this out for now and we can add it in another PR :)